### PR TITLE
add detail view

### LINKED ISSUE #13: 13. As a user, I want to view details of my purchases to better understand my purchase patterns.
AC:

* From the shopping list, user can tap to navigate to a detailed view of the item
* Detail view includes the following information:
  * Name of the item
  * Last purchased date
  * Estimated number of days until the next purchase
  * Number of times the item has been purchased
* When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view

### PR DESCRIPTION
**Pull request by sicunchen**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

In the List.js we added a new piece of state modalData to toggle the visibility of the detail modal. We also created a new component DetailModal.js that renders the content of the selected item. The related modal style is in App.css.

## Related Issue

closes #13 
## Acceptance Criteria

* From the shopping list, user can tap to navigate to a detailed view of the item
* Detail view includes the following information:
  * Name of the item
  * Last purchased date
  * Estimated number of days until the next purchase
  * Number of times the item has been purchased
* When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2020-09-19 at 14 30 06

### After

Screen Shot 2020-09-19 at 14 31 05

### CODE REVIEW

**Review by fab33150 — APPROVED** [review]
Great work! The modal and everything looks wonderful

File: src/App.css [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+  background-color: rgba(0, 0, 0, 0.25);
+}
+
+.modal_content {
+  position: absolute;
+  background-color: white;
+  position: abolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
```
Comment (yvesgurcan) [inline src/App.css:17]: As mentioned during the demo, I love the combination of `top`/`left` and `transform`. That's a clever way to make sure the modal will always be centered. How does it look on a mobile screen, though? Modals can be tricky on smaller devices.

File: src/DetailModal.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  const { name, nextPurchase, numberOfPurchases, lastPurchased } = data;
+  const handleClick = () => {
+    toggle(null);
+  };
+
+  return (
+    <div className="modal">
+      <div className="modal_content">
+        <span className="close" onClick={handleClick}>
+          &larr;
```
Comment (yvesgurcan) [inline src/DetailModal.js:13]: Yay! HTML code.

File: src/List.js [hunk 3]
```diff
@@ -66,42 +68,57 @@ const List = () => {
...
-                Delete
-              </button>
-            </li>
-          ))}
-        </ul>
-      )}
-      <NavLinks />
-    </div>
+    <Fragment>
+      {modalData && <DetailModal data={modalData} toggle={setModalData} />}
```
Comment (yvesgurcan) [inline src/List.js:81]: Well done! I like to use the `&&` trick to display a component conditionally.

**Review by yvesgurcan — APPROVED** [review]
This code is solid. Amazing job!

**Pull request merged**